dags/s3_xml_import_pipeline.py

```diff
@@ -0,0 +1,230 @@
+    )
+    dag_context = context["ti"]
+    data_config_dict = dag_context.xcom_pull(
+        key="data_config",
+        task_ids="Should_Remaining_Tasks_Execute"
+    )
+    data_config = eJPXmlDataConfig(data_config_dict, dep_env)
+
+    obj_pattern_with_latest_dates = (
+        get_stored_state(
+            data_config,
+            get_default_initial_s3_last_modified_date()
+        )
+    )
+    hook = S3HookNewFileMonitor(
+        aws_conn_id=named_literals.DEFAULT_AWS_CONN_ID,
+        verify=None
+    )
+    new_s3_files = hook.get_new_object_key_names(
+        obj_pattern_with_latest_dates,
+        data_config.s3_bucket
+    )
+    for object_key_pattern, matching_files_list in new_s3_files.items():
+        sorted_matching_files_list = (
+            sorted(
+                matching_files_list,
+                key=lambda file_meta:
+                file_meta[
+                    named_literals.S3_FILE_METADATA_LAST_MODIFIED_KEY
+                ]
+            )
+        )
+
+        for matching_file_metadata in sorted_matching_files_list:
+            etl_ejp_xml_zip(
+                data_config,
+                matching_file_metadata.get(
+                    named_literals.S3_FILE_METADATA_NAME_KEY
+                )
+            )
+            updated_obj_pattern_with_latest_dates = (
+                update_object_latest_dates(
+                    obj_pattern_with_latest_dates,
+                    object_key_pattern,
+                    matching_file_metadata.get(
+                        named_literals.S3_FILE_METADATA_LAST_MODIFIED_KEY
+                    )
+                )
+            )
+            update_state(
+                updated_obj_pattern_with_latest_dates,
+                data_config.state_file_bucket,
+                data_config.state_file_object
+            )
+
+
+def get_default_initial_s3_last_modified_date():
+    return os.getenv(
+        INITIAL_S3_XML_FILE_LAST_MODIFIED_DATE_ENV_NAME,
+        DEFAULT_INITIAL_S3_XML_FILE_LAST_MODIFIED_DATE
+    )
+
+
+SHOULD_REMAINING_TASK_EXECUTE = ShortCircuitOperator(
+    task_id='Should_Remaining_Tasks_Execute',
+    python_callable=is_dag_etl_running,
+    dag=S3_XML_ETL_DAG)
+
+
+NEW_S3_FILE_SENSOR = S3NewKeyFromLastDataDownloadDateSensor(
+    task_id='s3_key_sensor_task',
+    poke_interval=60 * 5,
+    timeout=60 * 60 * 24 * 1,
+    retries=0,
+    state_info_extract_from_config_callable=get_stored_state,
+    default_initial_s3_last_modified_date=(
+        get_default_initial_s3_last_modified_date()
+    ),
+    dag=S3_XML_ETL_DAG,
+    deployment_environment=os.getenv(
+        DEPLOYMENT_ENV_ENV_NAME,
+        DEFAULT_DEPLOYMENT_ENV_VALUE
+    )
+)
+
+
+LOCK_DAGRUN_UPDATE_PREVIOUS_RUNID = create_python_task(
+    S3_XML_ETL_DAG, "Update_Previous_RunID_Variable_Value_For_DagRun_Locking",
+    update_prev_run_id_var_val,
+)
+
+
+ETL_XML = create_python_task(
+    S3_XML_ETL_DAG, "Etl_XML",
```

Comment:
I am not sure how important the task names are.. just curious why `XML` is in all upper case where as `Etl` is not.